packages/next/client/normalize-trailing-slash.ts

```diff
@@ -11,7 +11,7 @@ export const normalizePathTrailingSlash = (path: string) => {
   }
 
   const { pathname, query, hash } = parsePath(path)
-  if (process.env.__NEXT_TRAILING_SLASH) {
```

Comment:
This same bug showed up in a few places and seemed related. `process.env.__NEXT_TRAILING_SLASH` is typed as `string | undefined` so if a user put `trailingSlash: false` in their next config it would give them the same behavior as `trailingSlash: true` since it shows up here as the string "false".